packages/worker/src/worker.ts

```diff
@@ -1051,9 +1019,8 @@ export class Worker {
             // Core has indicated that it will not return any more poll results, evict all cached WFs
             filter((state) => state !== 'POLLING'),
             first(),
-            map((): ContextAware<{ activation: coresdk.workflow_activation.WorkflowActivation; synthetic: true }> => {
+            map((): { activation: coresdk.workflow_activation.WorkflowActivation; synthetic: true } => {
```

Comment:
Not for this PR but I think we may be able to get rid of this code and just teardown the workflow runtime on shutdown.